libyul/optimiser/SyntacticalEquality.cpp

```diff
@@ -50,14 +50,18 @@ bool SyntacticallyEqual::expressionEqual(FunctionalInstruction const& _lhs, Func
 {
 	return
 		_lhs.instruction == _rhs.instruction &&
-		compareVector<Expression, &SyntacticallyEqual::operator()>(_lhs.arguments, _rhs.arguments);
+		containerEqual(_lhs.arguments, _rhs.arguments, [this](Expression const& _lhs, Expression const& _rhs) -> bool {
```

Comment:
Confusing: The parameters of the lambda have the same name as the parameters of the function.

Reply:
Disambiguated.